app/models/study_file.rb

```diff
@@ -1331,6 +1333,45 @@ def set_cluster_group_ranges
     end
   end
 
+  # update clustering fragments to be called on updates to study_file
+  def call_ingest_on_updated_clustering_fragments
+    if self.ann_data_file_info.present? && self.ann_data_file_info.data_fragments_changed? && self.file_type == "AnnData" && (!is_reference_anndata? || is_reference_anndata? != nil)
```

Comment:
There are some convenience methods that will make this simpler, I think.  `StudyFile#is_reference_anndata?` also calls `is_anndata?`, and all `AnnData` files will already have an `AnnDataFileInfo` object initialized:
```suggestion
    if !!is_reference_anndata? && ann_data_file_info.data_fragments_changed?
```
That will cover the 4 checks you already had, and `!!` does the same thing in Ruby it does in Javascript.

Also (I know I used to be bad about this, and you probably copied stuff of mine from elsewhere), you don't need to call `self` in Ruby unless you're doing a direct assignment in an initializer or need to pass a reference.  Any method/attribute can be called directly  - `file_type` and `is_reference_anndata?` are the same as `self.file_type` and `self.is_reference_anndata?` - and the `self` is implicit, depending on the method scope.